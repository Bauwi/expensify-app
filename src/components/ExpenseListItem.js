import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

export const ExpenseListItem = ({ id, description, amount, createdAt}) => {
  return (
    <div>
      <Link to={`/edit/${id}`}><h3>{description}</h3></Link>
      <p>
        am: {amount}
        Amount: {numeral(amount / 100).format('$0,0.00')} // 
        Created At: {moment(createdAt).format('MMMM Do, YYYY')}</p>
    </div>
  )
}

export default ExpenseListItem;
