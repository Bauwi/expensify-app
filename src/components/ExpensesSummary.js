import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotalAmount from '../selectors/expenses-total';
import selectExpensesCount from '../selectors/expenses-count';
import numeral from 'numeral';

export const ExpensesSummary = ({expensesTotalCount, expenseCount, expensesTotalAmount}) => {
  const expensesWord = expenseCount <= 1 ? "expense" : 'expenses';
  const expensesWordHidden = expensesTotalCount - expenseCount <= 1 ? "expense" : 'expenses';
  const formattedExpensesTotalAmount = numeral(expensesTotalAmount / 100).format('$0,0.00');
  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">Viewing <span>{expenseCount}</span> {expensesWord} totalling <span>{formattedExpensesTotalAmount}</span></h1>    
        <h3 className="page-header__subtitle"><span>{expensesTotalCount - expenseCount}</span> hidden {expensesWordHidden}</h3>
        <div className="page-header__actions">
          <Link className="button" to="/create">Add Expense</Link>
        </div>
      </div>
    </div>
  )
}

const mapStateToProps = (state) => {
  return {
    expensesTotalCount: selectExpensesCount(state.expenses),
    expenseCount: (selectExpenses(state.expenses, state.filters)).length,
    expensesTotalAmount: selectExpensesTotalAmount(selectExpenses(state.expenses, state.filters))
  }
}

export default connect(mapStateToProps)(ExpensesSummary);