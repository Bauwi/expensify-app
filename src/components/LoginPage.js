import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

 
export const LoginPage = ({startLoginGoogle, startLoginGithub}) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Expensify</h1>
      <p>Manage your expenses.</p>
      <button className="button button--login" onClick={startLoginGoogle}><i className="fa fa-google"></i> Login With Google</button> 
      <button className="button button--login" onClick={startLoginGithub}><i className="fa fa-github"></i> Login With GitHub</button>   
      
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLoginGoogle: () => dispatch(startLogin('google')),
  startLoginGithub: () => dispatch(startLogin('github'))
});

export default connect(undefined, mapDispatchToProps)(LoginPage);