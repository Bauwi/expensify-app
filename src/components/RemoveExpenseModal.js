import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import Modal from 'react-modal';
import { startRemoveExpense } from '../actions/expenses';

export class RemoveExpenseModal extends Component {
  onRemoveExpense = () => {
    this.props.startRemoveExpense({id: this.props.id});
    this.props.history.push('/dashboard');
  }


  render() {
    return (
      <div>
      <Modal
        isOpen={this.props.isOpen}
        onRequestClose={this.props.handleModalClose}
        contentLabel="Selected Option"
        closeTimeoutMS={200}
        className="modal"
      >
      <h3 className="modal__title">You are about to delete this expense. </h3>
      <h3 className="modal__title">Do you want to confirm ?</h3>
      <div>
        <button className="button button--danger button--aside" onClick={this.onRemoveExpense}>Remove</button>      
        <button className="button button--aside" onClick={this.props.handleModalClose}>Cancel</button>
      </div>
     </Modal>
    </div>
    )
  }
}

const mapStateToProps = (state, props) => {
  return {};
};

const mapDispatchToProps = (dispatch, props) => {
  return {
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
  }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(RemoveExpenseModal));
