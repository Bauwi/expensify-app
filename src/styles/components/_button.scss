.button {
  background-color: $blue;
  border: none;
  border-radius: 0;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;  
  line-height: 1;
  outline: none;
  padding: $s-size $m-size;
  text-decoration: none;
}

.button--login {
  width: 22rem;
  margin-bottom: $s-size;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888888;
}