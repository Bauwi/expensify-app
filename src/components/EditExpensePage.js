import React, { Component } from "react";
import { connect } from 'react-redux';

import ExpenseForm from './ExpenseForm';
import RemoveExpenseModal from './RemoveExpenseModal';

import { startEditExpense, startRemoveExpense } from '../actions/expenses';


export class EditExpensePage extends Component {
  state = {
    isOpen: false
  }

  handleModalOpen = () => {
    this.setState(() => ({
      isOpen: true
    }))
  }

  handleModalClose = () => {
    this.setState(() => ({
      isOpen: false
    }))
  }

  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/dashboard');
  }

  render() {
    return(
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm 
            onSubmit={this.onSubmit}
            expense={this.props.expense}
          />
          <button className="button button--secondary" onClick={this.handleModalOpen}
          >
            Remove Expense
          </button>
          <RemoveExpenseModal 
            id={this.props.expense.id}
            isOpen={this.state.isOpen}
            handleModalClose={this.handleModalClose}
          />
        </div>
    </div>
    )
  }
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => {
      return expense.id === props.match.params.id;
    })
  };
};

const mapDispatchToProps = (dispatch, props) => {
  return {
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
