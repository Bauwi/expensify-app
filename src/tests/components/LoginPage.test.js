import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should render LoginPage correctly', () => {
  const wrapper = shallow(<LoginPage startLoginGoogle={() => {}} startLoginGithub={() => {}}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLoginGoogle on Login With Google click', () => {
  const startLoginGoogle = jest.fn();
  const startLoginGithub = jest.fn();
  const wrapper = shallow(<LoginPage startLoginGoogle={startLoginGoogle} startLoginGithub={startLoginGithub}/>);
  wrapper.find('button').at(0).simulate('click');
  expect(startLoginGoogle).toHaveBeenCalled();
});

test('should call startLoginGithub on Login With Github click', () => {
  const startLoginGoogle = jest.fn();
  const startLoginGithub = jest.fn();
  const wrapper = shallow(<LoginPage startLoginGoogle={startLoginGoogle} startLoginGithub={startLoginGithub}/>);
  wrapper.find('button').at(1).simulate('click');
  expect(startLoginGithub).toHaveBeenCalled();
});