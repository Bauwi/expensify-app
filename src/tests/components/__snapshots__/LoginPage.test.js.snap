// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <p>
      Manage your expenses.
    </p>
    <button
      className="button button--login"
      onClick={[Function]}
    >
      <i
        className="fa fa-google"
      />
       Login With Google
    </button>
    <button
      className="button button--login"
      onClick={[Function]}
    >
      <i
        className="fa fa-github"
      />
       Login With GitHub
    </button>
  </div>
</div>
`;
